geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -113,7 +113,7 @@
 public class LocatorClusterManagementService implements ClusterManagementService {
   @VisibleForTesting
   // the dlock service name used by the CMS
-  static final String CMS_DLOCK_SERVICE_NAME = "CMS_DLOCK_SERVICE";
+  public static final String CMS_DLOCK_SERVICE_NAME = "CMS_DLOCK_SERVICE";
```

Comment:
Since "LocatorClusterManagementService" is central to the ClusterConfigurationService; will it be better to have the cms-dlock service created in this class; and have helper method to lock and unlock. Assumption is when the lock is requested the Locator Management service will be up and running.

lockConfigForUpdate() {}
unLockConfigForUpdate() {}

Reply:
I believe it is created by this class, see `getCmsDlockService` in this class

Reply:
In this PR. the Dlock is getting created in "OnlineCommandProcessor.java"; I don't have latest develop, my guess it may be getting created in rest endpoint too...

Reply:
they both call `DLockService.getOrCreateService`, if it's already created, it will just get it. So, I guess whoever get to it first will create it.

Reply:
Can that creating (getOrCreateService ) be done in one place, in "LocatorClusterManagementService" which seems to be right place for it...

Reply:
Then OnlineCommandProcessor needs to get ahold of `LocatorClusterManagementService` and get the dLock service from there? What if CMS is not started? Gfsh still needs to have this dLockService.

Reply:
My understanding was, we have centralized configuration-management class, that is used by both gfsh and rest end-point to make the configuration changes, and I thought it was LocatorClusterManagementService, I guess I am wrong...

Is it "InternalConfigurationPersistenceService" where both gfsh and end-point make the configuration changes...If so, can the dlock creation and unlocking/unlocking helper method be added there.

I am trying to see if we can have the lock creation in one place, rather than in many clients/apis that make the configuration changes...

Reply:
Yes, the `InternalConfigurationPersistenceService` is used by both rest and gfsh to persist the configuration change in the region, and it already has its own dlockService to make sure update to the region is thread safe. 

What are make thread-safe in here is not just "update to the region", it's the "update to the server sand then update the region", which is a level above the persistence layer.  Both rest and gfsh does this and so both needs the same dlockService.